src/lib/posts/scaling-for-millions.md

```diff
@@ -0,0 +1,154 @@
+Suppose our website attracts a lot more users during holiday season and not as much rest of the year. With Scaled-up architecture, we will be fine during the high trafic season but what happens to the extra server resources rest of the times? We will still be paying for the entire thing.  
+This also introduces 'Single Point Of Failure(SPOF)' another concept we should talk about.
+
+  **SPOF** or Single Point Of Failure is a component in an architechture that has a capability of f'ing up entire system. It says - 
+``
+If I go down, Imma take everyone with me.
+``
+
+Scaling up can still be a solution to some senarios or some businesses. It really depends on the nature of application and how important it is for the business to be running 24x7(online).
+- server is your pet
+- modify the diagrams
+
+ ### Horizontal Scaling or Scaling out
+
+Now for the favorite kind of scaling which is cheaper, efficient and easier to scale down. Scaling out simply refers to adding more servers at need. With increased traffic during holiday season, we will simply add more servers to our web-tier. If one of the servers go down the traffic will be directed to the next available server and in mean time we will spin-up a new server to replace the dead one thus avoiding SPOF.
+Similarly, when the traffic goes back to normal, we will remove the extra servers(scale down). We only pay for servers for the time we use them. 
+
+- you don't really care if one of the servers do down
+- treat them like cattle
+- want to scale on large level and all the time you need cattle not pets
+- don't take it personally
+
+## Load Balancer
+
+To be able to distribute the traffic equally to available servers, we need a component called Load Balancer. Load Balancer sits between user and servers(server group) and distributes data across the servers to keep the load on them consistent. Another good thing about laod balancer, it communicates with the servers through a private IP which means the user(malicious ones) cannot get their hands on our servers directly
+
+<div class="not-prose flex flex-col lg:flex-row justify-start items-center gap-10">
+<img src="/scaling/multi-server-with-db.svg" alt="" width="800px"/>
+<ol class="flex flex-col max-w-[600px] gap-2 text-lg"> 
+    <li>1. User searches for our website </li>
+    <li>2. DNS resolves the name(domain) to its corresponding IP address which user will use to make request to the server </li>
+    <li>3. User makes request to the server which goes through Load Balancer</li>
+    <li>4. Load Balancer decides which server to route the request to</li>
+    <li>5. Server queries the database and recieves the response </li>
+    <li>6. Server sends the data back to Load Balancer</li>
+    <li>7. Load Balancer serves it back to user</li>
+</ol>
+</div>
+
+## Database Replication
+
+Winter is coming! so are Holidays. Are user-base is expanding like a balloon but we are ready with our scaled-out web-tier. The users will be served!
```

Comment:
@guacamole typo here `Are user-base is expanding` should be `Our user-base is expanding` here.

Also, maybe swap the word `transaction` with `interaction (transaction in database world)`

Make `SPOF` here bold